### PR TITLE
fix misc issues

### PR DESCRIPTION
**Pull request by jackschofield23**
Signed-off-by: Jack Schofield <jack.schofield@answerdigital.com>

### Description

Fixes # .

Fixes a few problems spotted when trying to run the argo task via the workflow manager

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] All tests passed locally.
- [ ] [Documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments) included/updated.

### CODE REVIEW

File: src/TaskManager/TaskManager/appsettings.Development.json [hunk 1]
```diff
@@ -27,9 +28,9 @@
       "serviceAssemblyName": "Monai.Deploy.Storage.MinIO.MinIoStorageService, Monai.Deploy.Storage.MinIO",
       "settings": {
         "endpoint": "localhost:9000",
-        "accessKey": "minioadmin",
```
Comment (neildsouth) [inline src/TaskManager/TaskManager/appsettings.Development.json:30]: Think changing this here will break the mwm-developer-setup.md walkthough, the argo setup sets the user/pass on minio as minioadmin / minioadmin

File: src/TaskManager/Plug-ins/Argo/ArgoPlugin.cs [hunk 2]
```diff
@@ -635,7 +635,7 @@ private Messaging.Common.Storage GenerateIntermediaryArtifactStore(string artifa
             }
 
             var storageInfo = Event.IntermediateStorage.Clone() as Messaging.Common.Storage;
-            storageInfo!.RelativeRootPath = $"{storageInfo.RelativeRootPath}/{{{{ workflow.name }}}}/{artifactName}";
+            storageInfo!.RelativeRootPath = $"{storageInfo.RelativeRootPath}/{artifactName}";
```
Comment (mocsharp) [inline src/TaskManager/Plug-ins/Argo/ArgoPlugin.cs:638]: Just curious why remove the workflow name here

File: src/WorkflowManager/WorkflowExecuter/Common/ArtifactMapper.cs [hunk 3]
```diff
@@ -92,11 +106,11 @@ private static bool TrimArtifactVariable(string valueString, out string variable
             return true;
         }
 
-        private async Task<KeyValuePair<string, string>> ConvertVariableStringToPath(Artifact artifact, string variableString, string workflowInstanceId, string payloadId, string bucketId, bool shouldExistYet)
+        private async Task<KeyValuePair<string, string>> ConvertVariableStringToPath(Artifact artifact, string variableString, string workflowInstanceId, string payloadId, string bucketId, bool shouldExistYet, string suffix = "")
         {
             if (variableString.StartsWith("context.input.dicom", StringComparison.InvariantCultureIgnoreCase))
             {
-                return await VerifyExists(new KeyValuePair<string, string>(artifact.Name, $"{payloadId}/dcm/"), bucketId, shouldExistYet);
+                return await VerifyExists(new KeyValuePair<string, string>(artifact.Name, $"{payloadId}/dcm{suffix}"), bucketId, shouldExistYet);
```
Comment (mocsharp) [inline src/WorkflowManager/WorkflowExecuter/Common/ArtifactMapper.cs:113]: could you please also update the documentation with a few examples?

File: src/TaskManager/TaskManager/appsettings.Development.json (hunk 1 above)
Comment (jackschofield23) [reply src/TaskManager/TaskManager/appsettings.Development.json]: Reverted

File: src/TaskManager/Plug-ins/Argo/ArgoPlugin.cs (hunk 2 above)
Comment (jackschofield23) [reply src/TaskManager/Plug-ins/Argo/ArgoPlugin.cs:638]: This is due to us resolving the output artifacts before the argo task has been created, and the workflow name is only assigned when it has been created in argo, making it more complicated to include it in the path.

File: src/WorkflowManager/WorkflowExecuter/Common/ArtifactMapper.cs (hunk 3 above)
Comment (jackschofield23) [reply src/WorkflowManager/WorkflowExecuter/Common/ArtifactMapper.cs:113]: Added some docs

File: src/TaskManager/Plug-ins/Argo/ArgoPlugin.cs (hunk 2 above)
Comment (jackschofield23) [reply src/TaskManager/Plug-ins/Argo/ArgoPlugin.cs:638]: Marked as resolved as we need this merged but if there's any issues feel free to message/raise them in standup on Tuesday

**Review by JoeBatt1989 — APPROVED** [review]

File: src/TaskManager/TaskManager/appsettings.Local.json [hunk 4]
```diff
@@ -1,10 +1,11 @@
 {
   "WorkloadManagerDatabase": {
-    "ConnectionString": "mongodb://root:rootpassword@localhost:30017",
```
Comment (neildsouth) [inline src/TaskManager/TaskManager/appsettings.Local.json:3]: is it possible to leave this file as is ? or go through and update the developer-setup accordingly :)

File: src/WorkflowManager/WorkflowExecuter/Common/ArtifactMapper.cs [hunk 5]
```diff
@@ -76,6 +78,18 @@ IStorageService storageService
...
+        private static string? GetArtifactSuffix(string valueString)
+        {
+            var variableStrings = valueString.Split("}");
+
+            if (variableStrings.Length < 2)
+            {
+                return null;
+            }
+
+            return variableStrings[1];
```
Comment (lillie-dae) [inline src/WorkflowManager/WorkflowExecuter/Common/ArtifactMapper.cs:90]: I'm trying to think is it possible to have multiple variables in this field?
might be worth doing variableStrings.Last() I think thats linq way of getting last element.

**Pull request merged**